config_test.go

```diff
@@ -0,0 +1,27 @@
+package psdb
+
+import (
+	"reflect"
+	"testing"
+)
+
+func TestConfigFromURL(t *testing.T) {
+	cases := []struct {
+		in  string
+		cfg Config
+	}{
+		{"mysql://foo:bar@example.com/", Config{"example.com", "foo", "bar"}},
+		{"mysql://foo@example.com/", Config{"example.com", "foo", ""}},
+		{"mysql://foo:bar@example.com:9999", Config{"example.com:9999", "foo", "bar"}},
+		{"", Config{}},
+	}
+
+	for _, c := range cases {
+		t.Run(c.in, func(t *testing.T) {
+			cfg := ConfigFromURL(c.in)
+			if !reflect.DeepEqual(c.cfg, cfg) {
```

Comment:
imo testify's `assert.Equal` makes this a little nicer

Reply:
We don't even have testify in here, and I sorta struggle with these things in public projects since that'll get added into the main `go.mod`. It's common that if we get there, people vendor smaller parts of the assertion libraries they need, so they don't become dependencies, for example: https://github.com/bufbuild/connect-go/tree/main/internal/assert

Reply:
Hmm good point. It's unfortunate we'd have to bring that in as a dep just for testing.

Reply:
Oh, RIP. https://github.com/planetscale/psdb/blob/main/auth/auth_test.go#L6

I guess I do use it already, so should just figure something out. But this is still WIP anyways. Since it's arleady there, I agree, and will swap to use that. That was a poor assumption on my part.